spec/message.abnf

```diff
@@ -1,18 +1,22 @@
-message = pattern / complex-message
+message = simple-message / complex-message
 
-complex-message = "{{" [s] *(declaration [s]) body [s] "}}"
+simple-message = [simple-start pattern]
+simple-start = text-start / text-escape / expression
 
-declaration = input-declaration / local-declaration
+pattern = *(text / expression)
+
+complex-message = *(declaration [s]) complex-body
+declaration = input-declaration / local-declaration / reserved-declaration
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
+reserved-declaration = "." reserved-body 1*expression
 
-body = quoted-pattern
-     / (selectors 1*([s] variant))
-
+complex-body = quoted-pattern
+             / (selectors 1*([s] variant))
 quoted-pattern = "{{" pattern "}}"
-pattern = *(text / expression)
+
 selectors = match 1*([s] expression)
-variant = when 1*(s key) [s] quoted-pattern
+variant = 1*(s key) [s] quoted-pattern
```

Comment:
Whitespace handling seems a bit off here; I think it's only mandatory _between_ keys.
```suggestion
variant = key *(s key) [s] quoted-pattern
```

Reply:
No, that's already covered by the containing `selectors 1*([s] variant)` production.